client/tck/src/main/java/tck/graphql/typesafe/HeaderBehavior.java

```diff
@@ -586,4 +592,81 @@ void shouldAddCharsetRequestAndResponseHeaders() {
         then(fixture.sentHeader("Content-Type")).hasToString("application/json;charset=utf-8");
         then(fixture.sentHeader("Accept")).hasToString("application/json;charset=utf-8");
     }
+
+    @GraphQLClientApi
+    interface NonStringHeadersApi {
+        String greeting(@Header(name = "foo") Long id);
+    }
+
+    @Test
+    void shouldAddNonStringHeader() {
+        fixture.returnsData("'greeting':'dummy-greeting'");
+        NonStringHeadersApi api = fixture.build(NonStringHeadersApi.class);
+
+        api.greeting(123L);
+
+        then(fixture.sentHeader("foo")).isEqualTo("123");
+    }
+
+    @Test
+    void shouldNotAddNullNonStringHeader() {
+        fixture.returnsData("'greeting':'dummy-greeting'");
+        NonStringHeadersApi api = fixture.build(NonStringHeadersApi.class);
+
+        api.greeting(null);
+
+        then(fixture.sentHeader("foo")).isNull();
+    }
+
+    @GraphQLClientApi
+    @Header(method = "foo")
```

Comment:
So the header name will default to the method name? Is that really useful? Http headers names in most cases contain dashes and start with an uppercase letter, which is not quite compatible with naming of methods in Java. I don't see how this is useful in the real world.

Reply:
We already convert exception names to kebab-case for the `error-code` field on the server side. I did just that here, too. I find it very convenient to write `@Header(method = "contentType")` and it turns out as `Content-Type`. Same thing for parameter names! WDYT?

Reply:
I still find that a little bit odd, but ok, I won't oppose :) 
One last ask: Could you update `docs/typesafe-client-headers.md` with this?

Reply:
Yes, sure will.

Reply:
done